### PR TITLE
lesson_1_5

### PR DESCRIPTION
**Pull request by ledarca**
Lesson 1.5: Hooks And Fragments

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,19 +1,37 @@
-import React, { useState } from 'react';
-import TodoList from './TodoList';
-import AddTodoForm from './AddTodoForm';
 import "./App.css";
+import React from 'react';
+import AddTodoForm from './AddTodoForm';
+import TodoList from './TodoList';
+
+function useSemiPersistentState() {
+  const [todoList, setTodoList] = React.useState(
```
Comment (JoHeyJo) [inline src/App.jsx]: If you want to clean up your code a little more can actually destructure React hooks from your `react` module removing the need to call a hook on the `React` class.

File: src/App.jsx [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
 function App() {
-  const [todoList, setTodoList] = useState([]); 
+  const [todoList, setTodoList] = useSemiPersistentState();
+
+  const addTodo = (newTodo) => {
+    setTodoList([...todoList, newTodo]);
+  };
 
   return (
     <div>
```
Comment (JoHeyJo) [inline src/App.jsx]: This `<div>` element works just fine but take a look at the instructions again you're missing an important component that was brought up in the lesson.

**Review by JoHeyJo — APPROVED** [review]
Nicely done! Great implementation of a custom hook! Just a few things to look over but after addressing them you can go ahead and merge to main. The suggestion about destructuring React hooks isn't required, just want to be sure that you are aware of this. Keep up the good work!

File: src/App.jsx (hunk 1 above)
Comment (ledarca) [reply src/App.jsx]: Yesterday I had that doubt, thanks for clarifying for me.

File: src/App.jsx (hunk 2 above)
Comment (ledarca) [reply src/App.jsx]: thanks I didn't realize, updated

**Pull request merged**